File: get_missing.js
const fs = require('fs');
const texFile = 'docs/Tesis_Maestría_Matemáticas_Aplicadas_UNAL_2024 pacho/01Seccion01.tex';
const bibFile = 'docs/Tesis_Maestría_Matemáticas_Aplicadas_UNAL_2024 pacho/ReferenciasRugosas.bib';

try {
    const tex = fs.readFileSync(texFile, 'utf8');
    const bib = fs.readFileSync(bibFile, 'utf8');

    const cites = new Set();
    const regexTex = /\\cite\{([^}]+)\}/g;
    let match;
    while ((match = regexTex.exec(tex)) !== null) {
        const keys = match[1].split(',').map(k => k.trim());
        keys.forEach(k => cites.add(k));
    }

    const bibKeys = new Set();
    const regexBib = /@\w+\s*\{([^,]+)/g;
    while ((match = regexBib.exec(bib)) !== null) {
        bibKeys.add(match[1].trim());
    }

    const missing = [...cites].filter(x => !bibKeys.has(x));

    fs.writeFileSync('out.txt', `Missing: ${missing.join(', ')}\nTotal found in tex: ${cites.size}\n`);
    console.log("Success");
} catch (e) {
    fs.writeFileSync('out.txt', e.toString());
    console.log("Error", e);
}

File: get_missing.py
import re
tex_file = r"d:\Documents\GitHub\ChordSpace\docs\Tesis_Maestría_Matemáticas_Aplicadas_UNAL_2024 pacho\01Seccion01.tex"
bib_file = r"d:\Documents\GitHub\ChordSpace\docs\Tesis_Maestría_Matemáticas_Aplicadas_UNAL_2024 pacho\ReferenciasRugosas.bib"

with open(tex_file, "r", encoding="utf-8") as f:
    tex = f.read()

with open(bib_file, "r", encoding="utf-8") as f:
    bib = f.read()

cites = set()
for match in re.finditer(r'\\cite\{([^}]+)\}', tex):
    keys = [k.strip() for k in match.group(1).split(',')]
    cites.update(keys)

bib_keys = set()
for match in re.finditer(r'@\w+\s*\{([^,]+)', bib):
    bib_keys.add(match.group(1).strip())

missing = cites - bib_keys

with open("out.txt", "w", encoding="utf-8") as f:
    f.write(f"Missing: {missing}\n")
    f.write(f"Total found: {len(cites)}\n")
